js/src/components/google-combo-account-card/connected-google-combo-account-card.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import AccountCard, { APPEARANCE } from '../account-card';
+import AccountDetails from './account-details';
+import AppSpinner from '../app-spinner';
+import Indicator from './indicator';
+import getAccountCreationTexts from './getAccountCreationTexts';
+import useAutoCreateAdsMCAccounts from '.~/hooks/useAutoCreateAdsMCAccounts';
+import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
+import useGoogleMCAccount from '.~/hooks/useGoogleMCAccount';
+import './connected-google-combo-account-card.scss';
+
+/**
+ * Renders a Google account card UI with connected account information.
+ * It will also kickoff Ads and Merchant Center account creation if the user does not have accounts.
+ */
+const ConnectedGoogleComboAccountCard = () => {
+	// Used to track whether the account creation ever happened.
+	const [ wasCreatingAccounts, setWasCreatingAccounts ] =
+		useState( undefined );
+
+	const { hasFinishedResolution: hasFinishedResolutionForCurrentAdsAccount } =
+		useGoogleAdsAccount();
+
+	const { hasFinishedResolution: hasFinishedResolutionForCurrentMCAccount } =
+		useGoogleMCAccount();
+
+	const { hasDetermined, creatingWhich } = useAutoCreateAdsMCAccounts();
+	const { text, subText } = getAccountCreationTexts( wasCreatingAccounts );
+
+	const accountDetailsResolved =
+		hasDetermined &&
+		hasFinishedResolutionForCurrentAdsAccount &&
+		hasFinishedResolutionForCurrentMCAccount;
+
+	const displayAccountDetails =
+		accountDetailsResolved && wasCreatingAccounts === null;
+
+	useEffect( () => {
+		if ( hasDetermined ) {
+			setWasCreatingAccounts( creatingWhich );
+		}
+	}, [ creatingWhich, hasDetermined ] );
+
+	if (
+		wasCreatingAccounts === undefined &&
+		( ! hasDetermined ||
+			! hasFinishedResolutionForCurrentAdsAccount ||
+			! hasFinishedResolutionForCurrentMCAccount )
+	) {
```

Comment:
@eason9487 If we remove `wasCreatingAccounts === undefined`, it will show Spinner second time when the accounts have been created, but the `mc/connection` or `ads/connection` is being requested, so to keep the UI consistent, this check is required.